src/dodal/devices/oav/pin_image_recognition/__init__.py

```diff
@@ -50,38 +51,30 @@ def __init__(self, prefix: str, name: str = ""):
         self._prefix: str = prefix
         self._name = name
 
-        self.triggered_tip = create_soft_signal_r(Tip, "triggered_tip", self.name)
-        self.triggered_top_edge = create_soft_signal_r(
+        self.triggered_tip, _ = soft_signal_r_and_setter(
+            Tip, "triggered_tip", self.name
+        )
+        self.triggered_top_edge, _ = soft_signal_r_and_setter(
             NDArray[np.uint32], "triggered_top_edge", self.name
         )
-        self.triggered_bottom_edge = create_soft_signal_r(
+        self.triggered_bottom_edge, _ = soft_signal_r_and_setter(
             NDArray[np.uint32], "triggered_bottom_edge", self.name
         )
         self.array_data = epics_signal_r(NDArray[np.uint8], f"pva://{prefix}PVA:ARRAY")
 
         # Soft parameters for pin-tip detection.
-        self.preprocess_operation = create_soft_signal_rw(int, "preprocess", self.name)
-        self.preprocess_ksize = create_soft_signal_rw(
-            int, "preprocess_ksize", self.name
-        )
-        self.preprocess_iterations = create_soft_signal_rw(
+        self.preprocess_operation = soft_signal_rw(int, "preprocess", self.name)
+        self.preprocess_ksize = soft_signal_rw(int, "preprocess_ksize", self.name)
+        self.preprocess_iterations = soft_signal_rw(
             int, "preprocess_iterations", self.name
         )
-        self.canny_upper_threshold = create_soft_signal_rw(
-            int, "canny_upper", self.name
-        )
-        self.canny_lower_threshold = create_soft_signal_rw(
-            int, "canny_lower", self.name
-        )
-        self.close_ksize = create_soft_signal_rw(int, "close_ksize", self.name)
-        self.close_iterations = create_soft_signal_rw(
-            int, "close_iterations", self.name
-        )
-        self.scan_direction = create_soft_signal_rw(int, "scan_direction", self.name)
-        self.min_tip_height = create_soft_signal_rw(int, "min_tip_height", self.name)
-        self.validity_timeout = create_soft_signal_rw(
-            float, "validity_timeout", self.name
-        )
+        self.canny_upper_threshold = soft_signal_rw(int, "canny_upper", self.name)
+        self.canny_lower_threshold = soft_signal_rw(int, "canny_lower", self.name)
+        self.close_ksize = soft_signal_rw(int, "close_ksize", self.name)
+        self.close_iterations = soft_signal_rw(int, "close_iterations", self.name)
+        self.scan_direction = soft_signal_rw(int, "scan_direction", self.name)
+        self.min_tip_height = soft_signal_rw(int, "min_tip_height", self.name)
+        self.validity_timeout = soft_signal_rw(float, "validity_timeout", self.name)
```

Comment:
Must: As @evalott100 has said below, the arguments here have changed and we're now actually making signals with an `initial_value` of e.g. "canny_upper".

Reply:
Should: For bonus points could we use the initial values as in `connect`? I attempted to do this before but hit https://github.com/bluesky/ophyd-async/issues/266, which shouldn't be a problem now